File: evidence/generated-systems/golang/governance-on-golang/governance-on-golang/internal/model/Policy.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// Policy Declaration
//==============================================================
type Policy struct {
    gorm.Model
     Title                                    string
    VersionLabel                                    string
    ApprovalDate                                                            time.Time
    NextReviewDate                                                            time.Time
    DocumentUrl                                                            string
    OrganizationId         *uint
    Organization           *Organization `gorm:"foreignKey:OrganizationId"`
     Owners           []Person `gorm:"foreignKey:OwnersFromPolicyId"`
     RelatedRequirements           []ComplianceRequirement `gorm:"foreignKey:RelatedRequirementsFromPolicyId"`
     Controls           []Control `gorm:"foreignKey:ControlsFromPolicyId"`
     Procedures           []Procedure `gorm:"foreignKey:ProceduresFromPolicyId"`
     Exceptions           []Exception_ `gorm:"foreignKey:ExceptionsFromPolicyId"`
     Attestations           []Attestation `gorm:"foreignKey:AttestationsFromPolicyId"`
    PolicyType                      PolicyType
    Status                      DocumentStatus

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/governance-on-rails/governanceonrails/app/models/policy.rb
class Policy < ApplicationRecord
  enum PolicyType: [:InformationSecurity, :DataProtection, :Ethics, :RecordsManagement, :RiskManagement, :Compliance, :Privacy, :AcceptableUse]
  enum Status: [:Draft, :InReview, :Approved, :Retired]


  composed_of :uRL,
    class_name: "URL",
    mapping: [
      %w[uRL_value value]
    ]

  has_many :Organization, class_name: 'Organization'
  has_many :Owners, class_name: 'Person'
  has_many :RelatedRequirements, class_name: 'ComplianceRequirement'
  has_many :Controls, class_name: 'Control'
  has_many :Procedures, class_name: 'Procedure'
  has_many :Exceptions, class_name: 'Exception_'
  has_many :Attestations, class_name: 'Attestation'

end
